tagstudio/src/core/library.py

```diff
@@ -2315,3 +2175,340 @@ def sort_fields(self, entry_id: int, order: list[int]) -> None:
+                for id in self._tag_strings_to_id_map[term]:
+                    cluster.add(id)
+                    cluster = cluster.union(set(self.get_tag_cluster(id)))
+                # For each of the Tag IDs in the term's ID cluster:
+                for t in cluster:
+                    # Assume that this ID from the cluster is not in the Entry.
+                    # Wait to see if proven wrong.
+                    failure_to_union_terms = True
+                    # If the ID actually is in the Entry,
+                    if t in entry_tags:
+                        # There wasn't a failure to find one of the term's cluster IDs in the Entry.
+                        # There is also no more need to keep checking the rest of the terms in the cluster.
+                        failure_to_union_terms = False
+                        # print(f"FOUND MATCH: {t}")
+                        break
+                    # print(f'\tFailure to Match: {t}')
+        # # failure_to_union_terms is used to determine if all terms in the query were found in the entry.
+        # # If there even were tag terms to search through AND they all match an entry
+        if all_tag_terms and not failure_to_union_terms:
+            return add_entry(entry)
+        else:
+            return None
+
+    def add_entries_from_special_flags(
+        self,
+        special_flags: SpecialFlag,
+        entry_tags: list[int],
+        entry_authors: list[list[str]],
+        entry: Entry,
+    ):
+        if special_flags.only_untagged and not entry_tags:
+            return True
+        elif special_flags.only_no_author and not entry_authors:
+            return True
+        elif special_flags.only_empty and (not (entry.fields and entry.fields[0])):
+            return True
+        elif (
+            special_flags.only_missing
+            and (self.library_dir / entry.path / entry.filename).resolve()
+            in self.missing_files
+        ):
+            return True
+        return False
+
+    def remap_fields(self, entry: Entry) -> dict[str | int, str]:
+        entry_fields: dict = {}
+        for field in entry.fields:
+            for key, value in field.items():
+                key_mapped = self._field_id_to_name_map.get(int(key))
+                key_type = self.get_field_obj(key)["type"]
+                if (
+                    key_type != "tag_box"
+                    and key_mapped != "author"
+                    and key_mapped != "artist"
+                ):
+                    if key_mapped in entry_fields.keys():
+                        entry_fields[key_mapped].append(value)
+                    else:
+                        entry_fields[key_mapped] = [value]
+                else:
+                    if key_mapped in entry_fields.keys():
+                        entry_fields[key_mapped].extend(value)
+                    else:
+                        entry_fields[key_mapped] = value
```

Comment:
this is perfect use-case for `collections.defaultdict`:
`entry_dict = defaultdict(list)`
then you can do:
```suggestion
                    entry_fields[key_mapped].append(value)
                else:
                    entry_fields[key_mapped].extend(value)
```